Extension/src/LanguageServer/client.ts

```diff
@@ -1201,6 +1207,13 @@ export class DefaultClient implements Client {
             localizedStrings.push(lookupString(i));
         }
 
+        if (workspaceSettings.caseSensitiveFileSupport !== currentCaseSensitiveFileSupport.Value) {
+            resetDatabase = true;
+            if (workspaceSettings.caseSensitiveFileSupport) {
```

Comment:
Is this `IF` needed?  Shouldn't `currentCaseSensitiveFileSupport.Value` be updated every time we see that it changed?   As it is now, if the user has transitioned it from `true` to `false` (while our extension was not active, so we did not intercept it via a didChangeSettings event), then this will fail to update it to `false`.

Reply:
How would we detect a change in that scenario?

Reply:
I don't think @Colengms means to update it every time it changes (e.g. didChangeConfiguration). I believe he just means to update it when we create the language client.

And `workspaceSettings.caseSensitiveFileSupport` isn't a boolean, so this check will always return `true` anyway.

Reply:
Oh I understand, the if was originally there because the typescript complained about `workspaceSettings.caseSensitiveFileSupport` potentially being undefined. So the check is needed, unless its ok for me to removed the undefined type from the setting

Reply:
I thought the value we get from VSCode's `get<T>` API should never be undefined because we have a default value in `package.json` for this setting.  But if it can be undefined, then I understand why we'd want to keep the `if`.

Reply:
Be careful when using just the variable name itself as a boolean to detect if it's defined or not, in TypeScript.  That syntax will return false if it's defined but has a value of 0 or false.  Instead, always be explicit and use `var === undefined` or `var !== undefined`, unless your intention is to assume the variable is 0/false if undefined.

Bob is right that we should never receive an 'undefined' value for the setting, due to having a default in `package.json`.  However, the API is capable of returning `undefined`, so we need to check for that to make TypeScript happy and let us compile.

IMHO, it would be a good idea to resolve this setting into a boolean immediately after reading it, so you don't need to deal with it being a string or potentially undefined, after reading it from the settings class.